src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,104 @@
+    }
+
+    @Override
+    public void periodic() {
+        armIO.updateInputs(armInputs);
+        Logger.processInputs("Arm", armInputs);
+        updateMechanism();
+    }
+
+    boolean atAngle(Rotation2d targetAngle) {
+        return Math.abs(armInputs.anglePositionDegrees - targetAngle.getDegrees()) <= ArmConstants.ANGLE_TOLERANCE_DEGREES;
+    }
+
+    boolean atTargetElevatorPosition(double targetElevatorPositionMeters) {
+        return Math.abs(armInputs.elevatorPositionMeters - targetElevatorPositionMeters) <= ArmConstants.ELEVATOR_TOLERANCE_METERS;
+    }
+
+    boolean isElevatorOpening(double targetElevatorPositionMeters) {
+        return armInputs.elevatorPositionMeters < targetElevatorPositionMeters;
+    }
+
+    void generateAngleMotorProfile(Rotation2d targetAngle, double speedPercentage) {
+        angleMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ANGLE_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(armInputs.anglePositionDegrees, armInputs.angleVelocityDegreesPerSecond)
+        );
+
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    void generateElevatorMotorProfile(double targetElevatorPositionMeters, double speedPercentage) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ELEVATOR_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetElevatorPositionMeters, 0),
+                new TrapezoidProfile.State(armInputs.elevatorPositionMeters, armInputs.elevatorVelocityMetersPerSecond)
+        );
+
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            armIO.stopAngleMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        ArmConstants.TARGET_POSITION_LIGAMENT.setAngle(targetState.position);
+        armIO.setTargetAngleState(targetState);
+    }
+
+    void setTargetElevatorPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            armIO.stopElevatorMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        ArmConstants.TARGET_POSITION_LIGAMENT.setLength(targetState.position + ArmConstants.RETRACTED_ARM_LENGTH_METERS);
+        armIO.setTargetElevatorState(targetState);
+    }
+
+    private double getAngleMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastAngleMotorProfileGenerationTime;
+    }
+
+    private double getElevatorMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastElevatorMotorProfileGenerationTime;
+    }
+
+    @AutoLogOutput(key = "Arm/ArmMechanism")
```

Comment:
This is not where you use this, you want to put it next to the variable in the constants